### PR TITLE
[chore] Extract on button press logic

### PR DESCRIPTION
**Pull request by OscarBakker**
## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/utils/on-button-press.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2013 Gemeente Amsterdam
```
Comment (vdegraaf) [inline src/utils/on-button-press.ts:2]: 2013....

File: src/utils/on-button-press.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+export default function onButtonPress(
```
Comment (vdegraaf) [inline src/utils/on-button-press.ts:3]: test missing?

**Review by vdegraaf — COMMENTED** [review]
Nice improvement! But there are some other places where this should be implemented as well: 

- SelectOption
- StepByStepClickable
- FileInput
- Summary
- FilterCategory
- IncidentLayer

File: src/utils/on-button-press.ts (hunk 1 above)
Comment (OscarBakker) [reply src/utils/on-button-press.ts:2]: haha

**Comment by OscarBakker** [comment]
> Nice improvement! But there are some other places where this should be implemented as well:
> 
> * SelectOption
> * StepByStepClickable
> * FileInput
> * Summary
> * FilterCategory
> * IncidentLayer

alleen bij filminput niet nodig, want native support button

**Comment by vdegraaf** [comment]
@OscarBakker Why shouldn't these be changed? 
Screenshot 2023-08-02 at 11 30 02

**Comment by OscarBakker** [comment]
> @OscarBakker Why shouldn't these be changed? Screenshot 2023-08-02 at 11 30 02

wut, thought I had these allready transformed.welll....

**Review by vdegraaf — APPROVED** [review]

**Pull request merged**